### PR TITLE
Saver hook test

### PR DESCRIPTION
**Pull request by gnovis**

### CODE REVIEW

File: cxflow/tests/hooks/saver_hook_test.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        ('valid', collections.OrderedDict([
+            ('accuracy', 0.8),
+            ('loss', collections.OrderedDict([('mean', valid_loss_mean_val)])),
+        ]))
+    ])
+    return epoch_data
+
+
+class EmptyNet(AbstractNet):
+    """The net which raises ValueError if save method is called."""
```
Comment (petrbel) [inline cxflow/tests/hooks/saver_hook_test.py]: ValueError -> \`ValueError\`

File: cxflow/tests/hooks/saver_hook_test.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+from typing import Mapping, List
+import collections
+
+from cxflow.tests.test_core import CXTestCase
+from cxflow.hooks.saver_hook import SaverHook, BestSaverHook, AbstractHook
+from cxflow.nets.abstract_net import AbstractNet
+
+
+def _get_epoch_data(valid_loss_mean_val: float=3) -> AbstractHook.EpochData:
```
Comment (petrbel) [inline cxflow/tests/hooks/saver_hook_test.py]: missing docstring (at least very simple one)

File: cxflow/tests/hooks/saver_hook_test.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    def restore_fallback_class(self) -> str:
+        return ''
+
+
+class SaverHookTest(CXTestCase):
+    """Test case for SaverHook."""
+
+    def test_raise_on_save_failure(self):
+        """
+        Test raising an exception if on_save_failure
```
Comment (petrbel) [inline cxflow/tests/hooks/saver_hook_test.py]: on_save_failure -> \`on_save_failure\`

**Comment by gnovis** [comment]
@petrbel Sure, sorry, should be fixed, thanks.

**Pull request merged**